scripts/binary_to_array.py

```diff
@@ -0,0 +1,289 @@
+in the root namespace, it will be declared as a C array (using extern "C" if
+compiled in C++ mode).
+
+The actual size of $NAME_data is $NAME_length + 1, where it contains an extra
+0x00 at the end. When data is actually text, $NAME_data can be used as a valid C
+string directly.
+"""
+
+from os import path
+from re import sub
+import argparse
+import logging
+
+arg_parser = argparse.ArgumentParser()
+
+arg_parser.add_argument("--input", type=str, help="Input file containing binary data to embed")
+arg_parser.add_argument("--output_source", type=str,
+                    help="Output source file, defining the array data.")
+arg_parser.add_argument("--output_header", type=str,
+                    help="Output header file, declaring the array data.")
+arg_parser.add_argument("--array", type=str, help="Identifier for the array.")
+arg_parser.add_argument("--array_size", type=str, help="Identifier for the array size.")
+arg_parser.add_argument("--filename", type=str, help="Override file name in code.")
+arg_parser.add_argument("--filename_identifier", type=str, help="Where to put the filename.")
+arg_parser.add_argument("--header_guard", type=str,
+                    help="Header guard to #define in the output header.")
+arg_parser.add_argument("--cpp_namespace", type=str,
+                    help="C++ namespace to use. If blank, will generate a C array.")
+
+# How many hex bytes to display in a line. Each "0x00, " takes 6 characters, so
+# a width of 12 lets us fit within 80 characters.
+WIDTH = 12
+
+
+class Error(Exception):
+  """Exception raised by methods in this module."""
+  pass
+
+
+def header(header_guard, namespaces, array_name, array_size_name, fileid):
+  """Return a C/C++ header for the given array.
+
+  Args:
+    header_guard: Name of the HEADER_GUARD to define.
+    namespaces: List of namespaces, outer to inner.
+    array_name: Name of the array.
+    array_size_name: Name of the array size constant.
+    fileid: Name of the identifier containing the file name.
+
+  Returns:
+    A list of strings containing the C/C++ header file, line-by-line.
+  """
+
+  data = []
+  data.extend([
+      "// Copyright 2018 Google Inc. All Rights Reserved.",
+      "",
+      "#ifndef %s" % header_guard,
+      "#define %s" % header_guard,
+      "",
+      "#include <cstdlib>",
+      ""
+  ])
+  if namespaces:
+    data.extend([
+        "namespace %s {" % ns for ns in namespaces
+    ])
+  else:
+    data.extend([
+        "#if defined(__cplusplus)",
+        "extern \"C\" {",
+        "#endif  // defined(__cplusplus)"])
+
+  data.extend([
+      "",
+      "extern const size_t %s;" % array_size_name,
+      "extern const unsigned char %s[];" % array_name,
+      "extern const char %s[];" % fileid,
+  ])
+
+  data.extend([
+      ""
+  ])
+  if namespaces:
+    data.extend([
+        "}  // namespace %s" % ns for ns in namespaces
+    ][::-1])  # close namespaces in reverse order
```

Comment:
Alternatively: `"} //namespace %s" % ns for ns in reversed(namespaces)`

Reply:
Done, thanks.